pkg/rpc/core/blocks.go

```
@@ -163,55 +145,60 @@ func BlockByHash(ctx *rpctypes.Context, hash []byte) (*ctypes.ResultBlock, error
 func BlockResults(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlockResults, error) {
-	// Currently, this method returns an empty result as the original implementation was commented out.
-	// If block results become available, this implementation should be updated.
-	_ = normalizeHeight(heightPtr) // Use height to avoid unused variable error, logic depends on future implementation
-	return &ctypes.ResultBlockResults{}, nil
-	// Original commented-out logic:
 	// var h uint64
-	// if height == nil {
-	// 	 h = p.adapter.Store.Height(ctx)
+	// var err error
+	// if heightPtr == nil {
+	// 	h, err = env.Adapter.RollkitStore.Height(ctx.Context())
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
 	// } else {
-	// 	 h = uint64(*height)
+	// 	h = uint64(*heightPtr)
 	// }
-	// header, _, err := p.adapter.Store.GetBlockData(ctx, h)
+	// header, _, err := env.Adapter.RollkitStore.GetBlockData(ctx.Context(), h)
 	// if err != nil {
-	// 	 return nil, err
+	// 	return nil, err
 	// }
-	// resp, err := p.adapter.Store.GetBlockResponses(ctx, h)
+	// resp, err := env.Adapter.Store.GetBlockResponses(ctx.Context(), h)
 	// if err != nil {
-	// 	 return nil, err
+	// 	return nil, err
 	// }
-	// return &coretypes.ResultBlockResults{
-	// 	 Height:                int64(h), //nolint:gosec
-	// 	 TxsResults:            resp.TxResults,
-	// 	 FinalizeBlockEvents:   resp.Events,
-	// 	 ValidatorUpdates:      resp.ValidatorUpdates,
-	// 	 ConsensusParamUpdates: resp.ConsensusParamUpdates,
-	// 	 AppHash:               header.Header.AppHash,
+
+	// return &ctypes.ResultBlockResults{
+	// 	Height:                int64(h), //nolint:gosec
+	// 	TxsResults:            resp.TxResults,
+	// 	FinalizeBlockEvents:   resp.Events,
+	// 	ValidatorUpdates:      resp.ValidatorUpdates,
+	// 	ConsensusParamUpdates: resp.ConsensusParamUpdates,
+	// 	AppHash:               header.Header.AppHash,
 	// }, nil
+	return nil, nil // TODO: implement GetBlockResponses
```

Let's open a follow up issue with this. Also can we return an error to the user saying it not implemented instead of nil